### PR TITLE
Reorganizing MTC prerequisites, general cleanup, move upgrade modules to installation assembly

### PR DESCRIPTION
**Pull request by apinnick**
Changes:

- Prerequisites (installation, migration, launching console, upgrading) reorganized and standardized.
- Upgrade modules moved to "Deploying" assembly. Upgrading assembly deleted. This change requires redirects. Vikram will handle the redirects after peer review is completed.
- 4.x upgrade modules updated for manual Operator update. (They were overlooked when the installation procedure changed to 'manual' subscription policy.)
- Installing in restricted environment has been reorganized.
- "Configuring a replication repository" assembly title changed to "Configuring object storage"

4.5+

Peer review complete.

Previews:

Installation prerequisites

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html#migration-installing-mtc-on-ocp-4_migrating-3-4

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html#migration-installing-mtc-on-ocp-3_migrating-3-4

Installing in restricted environment

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html#migration-installing-mtc-on-ocp-4_disconnected-3-4

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html#migration-installing-mtc-on-ocp-3_disconnected-3-4

Migration prerequisites

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-applications-with-cam-3-4.html#migration-prerequisites_migrating-3-4

Console prerequisites

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/migrating-applications-with-cam-3-4.html#migration-launching-cam_migrating-3-4

Upgrades (each is different depending on whether the upgrade is on OCP 3 or OCP 4.x and whether on 4.x target or 4.x source cluster)

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_3_4/deploying-cam-3-4.html#upgrading-the-mtc_migrating-3-4

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_1_4/deploying-cam-4-1-4.html#upgrading-the-mtc_migrating-4-1-4

https://deploy-preview-31548--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_2_4/deploying-cam-4-2-4.html#upgrading-the-mtc_migrating-4-2-4

### CODE REVIEW

**Comment by apinnick** [comment]
@XinRedhat Please review the OAuth prerequisites that were added to the "Launching MTC console" section.

**Comment by apinnick** [comment]
@XinRedhat  This PR turned out larger than I expected! For QE, specifically, could you look at the prerequisites section (make sure that prereqs for installing MTC (short) and migrating applications (long) are OK? Also, have a look at the upgrade sections. You will have to check all 3 because the 4.x upgrades have slightly different procedures for upgrading on a source cluster (changes to migration controller). Thanks

**Comment by apinnick** [comment]
@vikram-redhat Since this is a rather complex PR, I requested your review at the same time as QE. Could you please set up the redirects for the Upgrade sections? The upgrade modules have been added to the "Installing and upgrading" assembly. The upgrade assembly has been removed.

**Comment by apinnick** [comment]
@vikram-redhat  Would like to merge fairly soon if possible.

**Comment by XinRedhat** [comment]
LGTM

**Review by xinredhat — COMMENTED** [review]
LGTM @apinnick

File: migration/migrating_3_4/deploying-cam-3-4.adoc [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
 
 :context: disconnected-3-4
 :disconnected-3-4:
-[id='installing-cam-operator_{context}']
-== Installing the {mtc-full} Operator in a restricted environment
+[id='installing-mtc_{context}']
+== Installing the {mtc-full} in a restricted environment
 
-You can install the {mtc-full} Operator with the Operator Lifecycle Manager (OLM) on an {product-title} {product-version} target cluster and manually on an {product-title} 3 source cluster.
+[IMPORTANT]
```
Comment (mburke5678) [inline migration/migrating_3_4/deploying-cam-3-4.adoc:32]: Do you want a _You can install the Migration Toolkit for Containers_ sentence here, like in the other sections?

File: migration/migrating_4_1_4/configuring-replication-repository-4-1-4.adoc [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [id='configuring-replication-repository-4-1-4']
-= Configuring a replication repository
+= Configuring object storage
```
Comment (mburke5678) [inline migration/migrating_4_1_4/configuring-replication-repository-4-1-4.adoc:2]: This seems confusing. It sounds like you are configuring storage for general reasons. I understand the change, as the user isn't really configuring a replication repository.
How about something like: _Configuring object storage for the replication repository_

File: migration/migrating_4_2_4/configuring-replication-repository-4-2-4.adoc [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [id='configuring-replication-repository-4-2-4']
-= Configuring a replication repository
+= Configuring object storage
```
Comment (mburke5678) [inline migration/migrating_4_2_4/configuring-replication-repository-4-2-4.adoc:2]: Same issue here as above

File: migration/migrating_4_2_4/deploying-cam-4-2-4.adoc [hunk 4]
```diff
@@ -1,72 +1,84 @@
...
 :context: migrating-4-2-4
 :migrating-4-2-4:
 
 toc::[]
 
-You can install the {mtc-full} Operator on your {product-title} {product-version} target cluster and 4.2 source cluster. The {mtc-full} Operator installs the {mtc-full} ({mtc-short}) on the target cluster by default.
+You can install the {mtc-full} Operator on your {product-title} {product-version} target cluster and 4.2 source cluster.
 
-[NOTE]
+{mtc-short} is installed on the target cluster by default. You can install {mtc-short} link:https://access.redhat.com/articles/5064151[on an {product-title} 3 cluster or on a remote cluster].
```
Comment (mburke5678) [inline migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:11]: _You can install the {mtc-full} Operator on your {product-title} {product-version} target cluster_
If the MTCO is installed on the target cluster by default, why do I need instructions to install if on the target cluster? I'm confused here.

File: migration/migrating_4_2_4/deploying-cam-4-2-4.adoc [hunk 5]
```diff
@@ -1,72 +1,84 @@
...
-[id='installing-cam-operator-prereq_{context}']
-=== Prerequisites
+[IMPORTANT]
+====
+You must install the same {mtc-short} version on all clusters.
+====
 
-* If you want to prune the default catalog and selectively mirror only a subset of Operators, install the xref:../../cli_reference/opm-cli.adoc#opm-cli[`opm` CLI].
+[id="using-operator-lifecycle-manager-on-restricted-networks_{context}"]
+=== Using Operator Lifecycle Manager on restricted networks
```
Comment (mburke5678) [inline migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:48]: I don't think you need this heading here. The paragraph that follows describe how to set up the OLM and environment to install the MTC. That isn't clear in the heading.

File: modules/migration-creating-migration-plan-cam.adoc [hunk 6]
```diff
@@ -9,40 +9,22 @@
...
-** The exposed registry route of the source cluster must be added to the cluster configuration by using either the {mtc-short} web console or the `exposedRegistryPath` parameter in the `MigCluster` CR manifest.
-
-* Direct volume migration:
-** The PVs to be migrated must be valid.
-** The PVs must be in a `bound` state.
-** The PVs must be copied by using the file system copy method.
-** You must configure a TCP proxy for Stunnel for direct volume migration in a proxy environment.
+* You must be logged in as a user with `cluster-admin` privileges on all clusters.
+* You must ensure that the same {mtc-short} version is installed on all clusters.
+* You must add the clusters and the replication repository to the {mtc-short} web console.
```
Comment (mburke5678) [inline modules/migration-creating-migration-plan-cam.adoc:16]: Should these two be past tense as the steps above show me how to do this? 
_You must ensure that the same {mtc-short} version was installed_
_You must have added_

File: modules/migration-creating-migration-plan-cam.adoc [hunk 7]
```diff
@@ -9,40 +9,22 @@
...
+* If you use the _move_ data copy method to migrate a persistent volume (PV), the source and target clusters must have uninterrupted network access to the remote volume.
+* If you use direct image migration, the `MigCluster` custom resource manifest of the source cluster must specify the exposed route of the internal image registry.
```
Comment (mburke5678) [inline modules/migration-creating-migration-plan-cam.adoc:18]: Should these be _If you plan to..._. Seems that would be a bit clearer

File: modules/migration-creating-migration-plan-cam.adoc [hunk 8]
```diff
@@ -51,24 +33,28 @@ You can use _direct image migration_ and _direct volume migration_ to migrate im
...
+* The *Copy* option copies the data from the PV of a source cluster to the replication repository and then restores the data on a newly created PV, with similar characteristics, in the target cluster.
+* The *Move* option unmounts a remote volume, for example, NFS, from the source cluster, creates a PV resource on the target cluster pointing to the remote volume, and then mounts the remote volume on the target cluster. Applications running on the target cluster use the same remote volume that the source cluster was using.
 
 . Click *Next*.
 . On the *Copy options* page, select a *Copy method* for each PV:
 
 * *Snapshot copy* backs up and restores data using the cloud provider's snapshot functionality. It is significantly faster than *Filesystem copy*.
 * *Filesystem copy* backs up the files on the source cluster and restores them on the target cluster.
++
+The file system copy method is required for direct volume migration.
```
Comment (mburke5678) [inline modules/migration-creating-migration-plan-cam.adoc:45]: _direct volume migration_
Does the average person doing the migration know what this is?

File: modules/migration-excluding-resources.adoc [hunk 9]
```diff
@@ -12,11 +12,11 @@ By default, the {mtc-short} excludes service catalog resources and Operator Life
 
 .Procedure
 
-. Edit the `MigrationController` CR manifest:
+. Edit the `MigrationController` custom resource manifest:
 +
 [source,terminal]
 ----
-$ oc edit migrationcontroller -n openshift-migration
+$ oc edit migrationcontroller <migration_controller -n openshift-migration
```
Comment (mburke5678) [inline modules/migration-excluding-resources.adoc:19]: Missing a `>` in `<migration_controller`

File: modules/migration-prerequisites.adoc [hunk 10]
```diff
@@ -8,15 +8,31 @@
...
+** The persistent volumes (PVs) must be valid.
+** The PVs must be bound to persistent volume claims.
+** If you copy the PVs by using the _move_ method, the clusters must have unrestricted network access to the remote volume.
+** If you copy the PVs by using the _snapshot_ copy method, the following prerequisites apply:
+*** The cloud provider must support snapshots.
+*** The volumes must have the same cloud provider.
+*** The volumes must be located in the same geographic region.
+*** The volumes must have the same storage class.
+
+* If you perform a direct volume migration in a proxy environment, you must configure an Stunnel TCP proxy.
```
Comment (mburke5678) [inline modules/migration-prerequisites.adoc:37]: s/an/a

File: modules/migration-upgrading-mtc-on-ocp-4.adoc [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  migration_ui: false
+----
+endif::[]
+ifdef::migrating-4-1-4,migrating-4-2-4[]
++
+You do not need to update the `MigrationController` CR manifest on the target cluster.
+
+. Click *Create*.
+endif::[]
+. Click *Workloads* -> *Pods* to verify that the {mtc-short} pods are running.
```
Comment (mburke5678) [inline modules/migration-upgrading-mtc-on-ocp-4.adoc:59]: Do we want a step here to make sure users who need to do this procedure are aware it is there? Something like:

. If you are upgrading the Migration Toolkit for Containers (MTC) version 1.3.x to 1.4, follow the steps in _Upgrading MTC 1.3 to 1.4_ in the following section.

**Comment by mburke5678** [comment]
@apinnick This looks great. I have a few questions/comments/suggestions. Overall, LGTM. Sorry it took me so long.

File: _topic_map.yml [hunk 12]
```diff
@@ -1928,11 +1928,9 @@ Topics:
     File: planning-migration-3-to-4
   - Name: About the Migration Toolkit for Containers
     File: migrating-application-workloads-3-4
-  - Name: Deploying the Migration Toolkit for Containers
+  - Name: Installing and upgrading the Migration Toolkit for Containers
     File: deploying-cam-3-4
-  - Name: Upgrading the Migration Toolkit for Containers
```
Comment (mburke5678) [inline _topic_map.yml:1933]: Why did you move the upgrade section to install? From a modular documentation standpoint, installing and upgrading are different user stories and **should** be in separate assemblies.

File: migration/migrating_4_2_4/deploying-cam-4-2-4.adoc (hunk 4 above)
Comment (apinnick) [reply migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:11]: Because you still have to install it, no matter what cluster it is installed on. It does not install itself. But if you want to isntall the MTC on a different cluster, you have to change the defaults. It may be obvious but we have to call it out because sometimes people assume that you can choose which cluster to install it during the installation. That is not the case. The user has to change the MigController CR manifest if MTC is installed on a non-target cluster.

Comment (apinnick) [reply migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:11]: I reworded it to make it clear that it is the migration-controller pod that runs on the target cluster by default.

File: modules/migration-creating-migration-plan-cam.adoc (hunk 6 above)
Comment (apinnick) [reply modules/migration-creating-migration-plan-cam.adoc:16]: No, because we do not assume that a module is always used in the same context. Therefore, there is no reason to change the tense.

File: modules/migration-creating-migration-plan-cam.adoc (hunk 7 above)
Comment (apinnick) [reply modules/migration-creating-migration-plan-cam.adoc:18]: Changing to "if you want to".

File: modules/migration-creating-migration-plan-cam.adoc (hunk 8 above)
Comment (apinnick) [reply modules/migration-creating-migration-plan-cam.adoc:45]: Yes. It is described in an earlier section and in the prerequisites.

File: modules/migration-upgrading-mtc-on-ocp-4.adoc (hunk 11 above)
Comment (apinnick) [reply modules/migration-upgrading-mtc-on-ocp-4.adoc:59]: It is problematic in modular docs to refer to other modules.

File: _topic_map.yml (hunk 12 above)
Comment (apinnick) [reply _topic_map.yml:1945]: Upgrade and installation are closely related. The upgrade assembly did not fit well into the upper structure. I checked with Vikram and he said that when he looks for installation procedures, he expects to find upgrade procedures in the same section. It's an interesting topic for discussion. While upgrade might be separate from installation from a mod docs POV, in the user's mind, they are related topics.

**Comment by apinnick** [comment]
@vikram-redhat 
This PR is ready for merge and CP. Thanks!

File: migration/migrating_4_2_4/deploying-cam-4-2-4.adoc (hunk 4 above)
Comment (mburke5678) [reply migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:11]: I got it. I do like the original sentence better; it seems more clear to me. 
_The Migration Toolkit for Containers Operator installs the Migration Toolkit for Containers (MTC) on the target cluster by default._
The way you have it sounds like the MTC is already installed by default. The default is *where* the MTC is installed; I read it that it was installed by default.

Comment (apinnick) [reply migration/migrating_4_2_4/deploying-cam-4-2-4.adoc:11]: Actually, MTC is installed on all the clusters. The difference is where the migration-controller pod is running.

**Comment by apinnick** [comment]
@vikram-redhat  Waiting for your merge and CP. Please set up the redirects. Thanks
